netbox_topology_views/static_dev/js/home.js

```diff
@@ -39,149 +34,112 @@ var options = {
-            let selected_devices = params.nodes;
-            for (let selected_device in selected_devices) {
-                let url = ""
-                if(String(selected_devices[selected_device]).startsWith("c")) {
-                    cid = selected_devices[selected_device].substring(1);
-                    url = "/circuits/circuits/" + cid + "/";
-                }
-                else if (String(selected_devices[selected_device]).startsWith("p")) {
-                    cid = selected_devices[selected_device].substring(1);
-                    url = "/dcim/power-panels/" + cid + "/";
-                }
-                else if (String(selected_devices[selected_device]).startsWith("f")) {
-                    cid = selected_devices[selected_device].substring(1);
-                    url = "/dcim/power-feeds/" + cid + "/";
-                }
-                else {
-                    url = "/dcim/devices/" + selected_devices[selected_device] + "/";
-                }
-                
-                window.open(url, "_blank");
-            }
-
-        });
-    }
-};
+// Render vis graph
+let graph = null // vis graph instance
+
+const container = document.querySelector('#visgraph')
+const coordSaveCheckbox = document.querySelector('#id_save_coords')
+;(function handleLoadData() {
+    if (!topologyData) return
 
-export function load_doc() {
-    if (document.readyState !== 'loading') {
-        iniPlotboxIndex();
-      } else {
-        document.addEventListener('DOMContentLoaded', iniPlotboxIndex);
+    function htmlTitle(text) {
+        const container = document.createElement('div')
+        container.innerHTML = text
+        return container
     }
-};
 
+    const nodes = new DataSet(
+        topologyData.nodes.map((node) => ({
+            ...node,
+            title: htmlTitle(node.title)
+        }))
+    )
+
+    const edges = new DataSet(
+        topologyData.edges.map((node) => ({
+            ...node,
+            title: htmlTitle(node.title)
+        }))
+    )
+    graph = new Network(container, { nodes, edges }, options)
+    graph.fit()
+
+    graph.on('dragEnd', function (params) {
+        if (!coordSaveCheckbox.checked) return
+
+        Promise.allSettled(
+            this.getPositions(params.nodes).map(async (node) => {
+                const res = await fetch(
+                    '/api/plugins/netbox_topology_views/save-coords/',
+                    {
+                        method: 'PATCH',
+                        headers: {
+                            'X-CSRFToken': csrftoken,
+                            Accept: 'application/json',
+                            'Content-Type': 'application/json'
+                        },
+                        body: JSON.stringify({
+                            node_id: node,
+                            x: node.x,
+                            y: node.y
+                        })
+                    }
+                )
```

Comment:
Please change to:

```
Promise.allSettled(
            
           params.nodes.map(async (node) => {
                const res = await fetch(
                    '/api/plugins/netbox_topology_views/save-coords/save_coords/',
                    {
                        method: 'PATCH',
                        headers: {
                            'X-CSRFToken': csrftoken,
                            'Accept': 'application/json',
                            'Content-Type': 'application/json'
                        },
                        body: JSON.stringify({
                            node_id: node,
                            x: this.getPositions(node)[node].x,
                            y: this.getPositions(node)[node].y
                        })
                    }
                )

Reply:
Otherwise the coords are not saved during the 'dragEnd' event

Reply:
After investigating the function `graph.getPositions(params.nodes)` returns an object like `{node_id: {x: ..., y: ...},...}` and have the function to map over object entries instead of calling `graph.getPosition(node)` on every node